src/quo2/components/avatars/user_avatar/view.cljs

```diff
@@ -0,0 +1,56 @@
+(ns quo2.components.avatars.user-avatar.view
+  (:require [clojure.string :as string]
+            [quo2.components.avatars.user-avatar.style :as style]
+            [quo2.components.markdown.text :as text]
+            [react-native.core :as rn]
+            [react-native.fast-image :as fast-image]))
+
+(defn- extract-initials
+  [full-name amount-initials]
+  (let [upper-case-first-letter (comp string/upper-case first)
+        names-list              (string/split full-name " ")]
+    (->> names-list
+         (map upper-case-first-letter)
+         (take amount-initials)
+         (string/join))))
+
+(defn initials-avatar
+  [{:keys [full-name size draw-ring?]}]
+  (let [font-size       (get-in style/sizes [size :font-size])
+        amount-initials (if (#{:xs :xxs :xxxs} size) 1 2)]
+    [rn/view {:accessibility-label :initials-avatar
+              :style               (style/initials-avatar size draw-ring?)}
+     [text/text
+      {:style  style/initials-avatar-text
+       :size   font-size
+       :weight :semi-bold}
+      (extract-initials full-name amount-initials)]]))
+
+(def valid-ring-sizes #{:big :medium :small})
+
+(defn user-avatar
+  "If no `profile-picture` is given, draws the initials based on the `full-name` and
+  uses `ring-background` to display the ring behind the initials when given. Otherwise,
+  shows the `profile-picture` which already comes with the ring drawn."
+  [{:keys [full-name status-indicator? online? size profile-picture ring-background]
+    :or   {status-indicator? true
+           online?           true
+           size              :big}}]
+  (let [full-name    (or full-name "empty name")
```

Comment:
We probably don't want to leak English words without translation. Is that the case here?

Reply:
Yes I think you are right @ilmotta. In the designs for profile input it uses the default value "Your Name" so we probably want this default value to be configurable.
<img width="438" alt="Screenshot 2023-03-06 at 11 59 00" src="https://user-images.githubusercontent.com/22799766/223104685-4301805f-7506-4c83-b9be-40fcf66eefee.png">
or check the designs
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=15647%3A187526&t=cgWRufQfp9yXxcqW-0

Reply:
although, I guess we can set that anyway and Empty name is just an example value. Probably it can be pulled out and moved to the preview screen 👍